blockwork/build/caching.py

```diff
@@ -48,17 +48,136 @@
 import hashlib
 import os
 from pathlib import Path
-from typing import Optional, TYPE_CHECKING
+import sys
+from types import ModuleType
+from typing import DefaultDict, Optional, TYPE_CHECKING
 if TYPE_CHECKING:
     from ..transforms import Transform
 from ..context import Context
+from ordered_set import OrderedSet as OSet
+import distutils.sysconfig
+import ast
+import site
+
 
 # Switch to never pull or push from the cache, but to instead compare the
 # hash with existing cache entries.
 # This will become a command option later...
 CACHE_CONSISTENCY_MODE = False
 
+class PyHasher:
```

Comment:
Is this something that might benefit from a liberal application of `lru_cache`?

Reply:
the `self.hash_map` dict will prevent work being duplicated